qa/tasks/cephfs/test_misc.py

```diff
@@ -168,7 +169,7 @@ def test_filtered_df(self):
                                                   '-f', 'json-pretty')
         _ = json.loads(out)
 
-        proc = self.mount_a.run_shell(['df', '.'])
+        proc = self.mount_a.run_shell(args=['df', '.'], stdout=StringIO())
```

Comment:
There is no sense to use stdout=StringIO, why just don't directly use remote.sh:
Instead of:
```
proc = self.mount_a.run_shell(args=['df', '.'], stdout=StringIO())
output = proc.stdout.getvalue()
```
why not just use:
```
output = self.mount_a.client_remote.sh(['sudo', 'df', self.mount_a.mountpoint])
```

Reply:
Dropping the run_shell() make sense, but IMO we'd better in one separate PR, because it will involve hundreds of line code, which is a little risky here. Locally I have enhanced this commit.

Reply:
Rishabh have fix the partially fixed it in https://github.com/ceph/ceph/pull/34595, and for the py3 compatible is fixing in https://github.com/ceph/ceph/pull/34781.